.github/workflows/pr-check.yml

```diff
@@ -0,0 +1,32 @@
+name: Test greentea-client
+
+on: [push, pull_request]
+
+jobs:
+  astyle:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install astyle
+      run: sudo apt-get install astyle
+
+    - name: Run astyle
+      run: |
+          git diff --name-only --diff-filter=d FETCH_HEAD..HEAD \
+            | ( grep '.*\.\(c\|cpp\|h\|hpp\)$' || true ) \
+            | while read file; do astyle -n --options=.astylerc "${file}"; done
+          git diff --exit-code --diff-filter=d --color
+
+  test-app:
+    runs-on: ubuntu-latest
```

Comment:
With separate container instances be used for astyle and test-app? Could there be performance impact compared to having all tests in one job?

Reply:
Separating them out allows github to parallelize the job runs, so the checks should run more quickly than if everything was in a single job.